### PR TITLE
Compile tools to executables.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: Makefile [hunk 1]
```diff
@@ -58,7 +58,7 @@ prefix ?= /opt/toit-sdk
 all: sdk
 
 .PHONY: sdk
-sdk: tools snapshots version-file
+sdk: tools toit_tools version-file
```
[the remark this answers is not in the document]
Comment (floitsch) [reply Makefile]: done.

File: Makefile [hunk 2]
```diff
@@ -176,7 +176,7 @@ pi: pi-sysroot
 	$(MAKE) CROSS_ARCH=raspberry_pi SYSROOT="$(CURDIR)/build/$(PI_CROSS_ARCH)/sysroot" all-cross
 
 # ESP32 VARIANTS
-SNAPSHOT_DIR = build/$(HOST)/sdk/snapshots
+SNAPSHOT_DIR = build/$(HOST)/sdk/toit_tools
```
[the remark this answers is not in the document]
Comment (floitsch) [reply Makefile]: done.

File: tests/cmake/CMakeLists.txt [hunk 3]
```diff
@@ -15,44 +15,77 @@
...
 # works in that case too.
 set(CMAKE_TEST_INPUT_DIR "${CMAKE_BINARY_DIR}/cmake_test/dir with spaces")
 set(CMAKE_TEST_INPUT "${CMAKE_TEST_INPUT_DIR}/input.toit")
 set(CMAKE_TEST_SNAPSHOT "${CMAKE_BINARY_DIR}/cmake_test/cmake.snapshot")
+set(CMAKE_TEST_EXE "${CMAKE_BINARY_DIR}/cmake_test/cmake.exe")
 
-ADD_TOIT_TARGET(
-  ${CMAKE_TEST_INPUT}
-  ${CMAKE_TEST_SNAPSHOT}
+ADD_TOIT_SNAP(
```
[the remark this answers is not in the document]
Comment (floitsch) [reply tests/cmake/CMakeLists.txt]: done.

File: tools/toit.cmake [hunk 4]
```diff
@@ -42,7 +42,7 @@ if (DEFINED EXECUTING_SCRIPT)
 endif()
 
 # Creates a custom command to build ${TARGET} with correct dependencies.
-function(ADD_TOIT_TARGET SOURCE TARGET DEP_FILE ENV)
+function(ADD_TOIT_SNAP SOURCE TARGET DEP_FILE ENV)
```
[the remark this answers is not in the document]
Comment (floitsch) [reply tools/toit.cmake]: done.

**Pull request merged**